### PR TITLE
Fixed: Admin TrackCategory

### PR DESCRIPTION
**Pull request by ShubhamMewara**
PR Fixes: #396 also,
Fixes: #397 

https://github.com/code100x/daily-code/assets/136337515/a6afe023-35d6-4e5d-8665-60b117989ebc

### CODE REVIEW

**Comment by kushaltolia** [comment]
Hey, how did you sign in as an admin

**Comment by ShubhamMewara** [comment]
> Hey, how did you sign in as an admin

either you can change it in db or remove the if statement in `layout.tsx`

File: packages/ui/src/EditTrackCard.tsx [hunk 1]
```diff
@@ -2,23 +2,46 @@
...
   const [description, setDescription] = useState(Track.description);
   const [image, setImage] = useState(Track.image);
   const [hidden, setHidden] = useState(Track.hidden);
+  const [selectedCategory, setSelectedCategory] = useState<string[]>(Track.categories.map((item) => item.category.id));
+
+
   function handleEdit(id: string) {
     if (isEditing) {
-      updateTrack(id, { title, description, image, hidden });
+      updateTrack(id, { id, title, description, image, hidden, selectedCategory});
```
Comment (hkirat) [inline packages/ui/src/EditTrackCard.tsx:31]: Would this work?
`selectedCategory` isn't a key in the `db.track.update` function

Have u tested to see if this works?

Comment (ShubhamMewara) [reply packages/ui/src/EditTrackCard.tsx:31]: it is key `const [selectedCategory, setSelectedCategory] = useState<string[]>(Track.categories.map((item) => item.category.id));`

Comment (ShubhamMewara) [reply packages/ui/src/EditTrackCard.tsx:31]: > Have u tested to see if this works?

ya I have tested it

Comment (hkirat) [reply packages/ui/src/EditTrackCard.tsx:31]: Can u update the type of the `updateTrack` function from any to the right type as well?
Over here - https://github.com/code100x/daily-code/blob/main/apps/web/components/utils.tsx#L62

Comment (ShubhamMewara) [reply packages/ui/src/EditTrackCard.tsx:31]: yes i will

Comment (ShubhamMewara) [reply packages/ui/src/EditTrackCard.tsx:31]: > Can u update the type of the `updateTrack` function from any to the right type as well? Over here - https://github.com/code100x/daily-code/blob/main/apps/web/components/utils.tsx#L62

it's done

File: apps/web/components/utils.tsx [hunk 2]
```diff
@@ -241,16 +243,46 @@ export async function createTrack(data: {
...
       where: {
         id: trackId,
       },
-      data,
+      data:{
+        id: data.id,
+        title: data.title,
+        description: data.description,
+        image: data.image,
+        hidden: data.hidden,
```
Comment (kushaltolia) [inline apps/web/components/utils.tsx:265]: Don't we need to add selectedCategory : data.selectedCategory

Comment (ShubhamMewara) [reply apps/web/components/utils.tsx:271]: No, we don't as I'm doing it below in the same function

File: packages/ui/src/EditTrackCard.tsx (hunk 1 above)
Comment (ShubhamMewara) [reply packages/ui/src/EditTrackCard.tsx:31]: hey, @hkirat 
I have fixed this https://github.com/code100x/daily-code/pull/398#issuecomment-2099042050 
the import error in `LessonView.tsx`
in a commit https://github.com/code100x/daily-code/pull/400/commits/75fd7b381ea64d0fd086786d0d6f13c809897886

**Pull request merged**